src/components/Item/VisualizationItem/Visualization/IframePlugin.js

```diff
@@ -0,0 +1,106 @@
+import { useD2 } from '@dhis2/app-runtime-adapter-d2'
+import postRobot from '@krakenjs/post-robot'
+import PropTypes from 'prop-types'
+import React, {
+    Suspense,
+    useRef,
+    useCallback,
+    useState,
+    useEffect,
+} from 'react'
+import { itemTypeMap } from '../../../../modules/itemTypes.js'
+import { getPluginOverrides } from '../../../../modules/localStorage.js'
+import { useUserSettings } from '../../../UserSettingsProvider.js'
+import classes from './styles/DataVisualizerPlugin.module.css'
+import VisualizationErrorMessage from './VisualizationErrorMessage.js'
+
+const IframePlugin = ({
+    filterVersion,
+    item,
+    style,
+    visualization,
+    dashboardMode,
+}) => {
+    const { d2 } = useD2()
+    // TODO replace d2 with useCachedDataQuery
+    const apps = d2.system.installedApps
+    const { userSettings } = useUserSettings()
+    const iframeRef = useRef()
+    const [error, setError] = useState(false)
+
+    const onError = () => setError(true)
+
+    useEffect(() => {
+        const listener = postRobot.on(
+            'getProps',
+            // listen for messages coming only from the iframe rendered by this component
+            { window: iframeRef.current.contentWindow },
```

Comment:
Any reason to be concerned that `iframRef` wouldn't exist yet?

Reply:
Once `Suspense` is removed I don't think so.

Reply:
Removed `Suspense`